### PR TITLE
Add resume functionality

### PR DESCRIPTION
**Pull request by mihaigalos**
This PR implements the `REST` FTP verb, enabling resume of partially downloaded files.

### CODE REVIEW

**Comment by mihaigalos** [comment]
Would a static review suffice, since we are essentially just sending the remote endpoint text commands?

File: src/ftp.rs [hunk 1]
```diff
@@ -288,6 +288,12 @@ impl FtpStream {
         self.read_response(status::CLOSING).await?;
         Ok(())
     }
+    /// Sets the byte from which the transfer is to be restarted.
+    pub fn restart_from(&mut self, offset: u64) -> crate::Result<()> {
```
Comment (dani-garcia) [inline src/ftp.rs]:
```suggestion
    pub async fn restart_from(&mut self, offset: u64) -> Result<()> {
```

File: src/ftp.rs [hunk 2]
```diff
@@ -288,6 +288,12 @@ impl FtpStream {
...
+        let rest_command = format!("REST {}\r\n", offset.to_string());
+        self.write_str(&rest_command)?;
```
Comment (dani-garcia) [inline src/ftp.rs]:
```suggestion
        self.write_str(&rest_command).await?;
```

File: src/ftp.rs [hunk 3]
```diff
@@ -288,6 +288,12 @@ impl FtpStream {
...
+        self.read_response(status::REQUEST_FILE_PENDING).map(|_| ())
```
Comment (dani-garcia) [inline src/ftp.rs]:
```suggestion
        self.read_response(status::REQUEST_FILE_PENDING).await.map(|_| ())
```

**Review by dani-garcia — CHANGES_REQUESTED** [review]
You need to declare the function as async and make sure to await the other async calls, otherwise it looks pretty straightforward.

**Review by dani-garcia — APPROVED** [review]

**Comment by dani-garcia** [comment]
Thanks!

**Comment by mihaigalos** [comment]
No problem. How will the version be bumped to crates.io ?
Would like to consume these changes through cargo.

**Comment by dani-garcia** [comment]
I made a release now with your changes and some other small changes I had pending, as 5.1.0.
I've also updated tokio_rustls, which is an API break if you are using the secure feature, released as 6.0.0

**Comment by mihaigalos** [comment]
Awesome. Thank you.

**Pull request merged**